src/Datadog.Trace/TracerSubProcessManager.cs

```diff
@@ -10,40 +11,74 @@ namespace Datadog.Trace
 {
     internal class TracerSubProcessManager
     {
-        private static Task _traceAgentMonitor;
-        private static Task _dogStatsDMonitor;
+        private static readonly List<SubProcessMetadata> SubProcesses = new List<SubProcessMetadata>()
+        {
+            new SubProcessMetadata()
```

Comment:
Maybe we can move these to `Tracer` and have `TracerSubProcessManager` be a general-purpose process manager. It might look something like:

```csharp
var processManager = new TracerSubProcessManager();
processManager.Add(traceAgentProcessDetails);
processManager.Add(statsdProcessDetails);
processManager.StartProcesses();
```
or maybe you pass the list of processes into the constructor.

Reply:
Can I call YAGNI?

Reply:
Sure you can!